pkg/volume/portworx/portworx.go

```diff
@@ -171,6 +172,24 @@ func (plugin *portworxVolumePlugin) newProvisionerInternal(options volume.Volume
 	}, nil
 }
 
+func (plugin *portworxVolumePlugin) RequiresFSResize() bool {
+	return false
+}
+
+func (plugin *portworxVolumePlugin) ExpandVolumeDevice(
+	spec *volume.Spec,
+	newSize resource.Quantity,
+	oldSize resource.Quantity) (resource.Quantity, error) {
+	glog.V(4).Infof("Expanding: %s from %v to %v", spec.Name(), oldSize, newSize)
+	err := plugin.util.ResizeVolume(spec, newSize, plugin.host)
```

Comment:
@gnufied the Portworx driver implementation on the server side is already idempotent. So invoking it multiple times will have the same effect as it internally checks if the volume has the size. So I don't need any logic here on the client side. Best to keep it simple.

Reply:
There is another reason for keeping this idempotent in Kubernetes rather than in Portworx server side. Lets say:

1. User created a PVC of 5GB size and is matched with a PV which is of 7GiB size.
2. Later on User edits the PVC to be 6GiB. Kuberenetes may still call `ExpandVolumeDevice`  even though underlying PV already matches the requested size.
3. What happens in portworx server side now? Will it shrink existing 7GiB disk to 6GiB or will it return an error?

Obviously shrinking will be bad. We never shrink volumes in Kubernetes. Returning an error will be bad too. Even if theoretically Portworx in Server side does the "right thing" and returns successful response - there is no guarantee that the behaviour will remain the same always.  That is why - it is important to not rely on Portworx behaviour in server side.

Reply:
@gnufied good point ! Updated the review with check to see if existing volume is already equal or greater than requested size.